massa-final-state/src/final_state.rs

```diff
@@ -159,11 +174,50 @@ impl FinalState {
             final_state
                 .db
                 .write()
-                .write_batch(batch, Some(recovered_slot));
+                .write_batch(batch, Default::default(), Some(recovered_slot));
         }
 
         final_state.last_slot_before_downtime = Some(recovered_slot);
 
+        // Check that MIP store is coherent with the network shutdown time range
+        // Assume that the final state has been edited during network shutdown
```

Comment:
I think this check can also be done after bootstrap (in the same place we check is_db_valid() ), because we bootstrap both the last_start_period and the last_slot_before_downtime fields that lets us compute everything.

Reply:
done